net-p2p/nicotine+/nicotine+-3.3.4.ebuild

```
@@ -16,18 +16,36 @@ S="${WORKDIR}/nicotine-plus-${PV}"
 LICENSE="GPL-3+"
 SLOT="0"
 KEYWORDS="amd64 ppc x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
 
 # NOTE: good link - https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/DEPENDENCIES.md
-BDEPEND="sys-devel/gettext" # TODO(setan): maybe add pycodestyle and pylint here if use test
+BDEPEND="
+	sys-devel/gettext
+	test? (
+		dev-python/pytest
+		|| (
+		   >=gui-libs/gtk-4.6.9[broadway]
+		   >=x11-libs/gtk+-3.22.20:3[broadway]
+		)
+	)
+"
 RDEPEND="
 	dev-python/pygobject:3[${PYTHON_USEDEP}]
-	|| ( >=gui-libs/gtk-4.6.9[introspection] >=x11-libs/gtk+-3.22.20:3[introspection] )
+	|| (
+		>=gui-libs/gtk-4.6.9[introspection]
+		>=x11-libs/gtk+-3.22.20:3[introspection]
+	)
 "
 
 distutils_enable_tests pytest
 
 DOCS=( AUTHORS.md NEWS.md README.md TRANSLATORS.md )
 
+src_prepare() {
+	# remove version tests violating network sandbox
+	rm "${S}"/pynicotine/tests/unit/test_version.py
```

```suggestion
	rm "${S}"/pynicotine/tests/unit/test_version.py || die
```
But there may be better ways to disable it:
https://projects.gentoo.org/python/guide/test.html#tests-requiring-internet-access
https://projects.gentoo.org/python/guide/test.html#skipping-problematic-tests

can **rm** if no easy way works.